### ISSUE TITLE
Seemingly random characters in response body

### ISSUE THREAD

**Issue by matthenning** [cpp-netlib/cpp-netlib#784]
I'm relatively new to C++ so please excuse me if I missed something obvious.

In my response bodies there are seemingly random characters before, after and sometimes in between the actual response body.

My test code:

```
using namespace boost::network;

ApiClient::ApiClient(std::string hostname, std::string endpoint) {
        http::client::options options;
        options.openssl_sni_hostname(hostname);
        http::client client(options);

        std::ostringstream url;
        url << "https://" << hostname << "/api/v1/" << endpoint;

        http::client::request request(url.str());
        request << header("Authorization", "Bearer ****");
        request << header("Accept", "application/json; charset=utf-8");
        request << header("Connection", "close");

        http::client::response r = client.get(request);
        std::cout << http::body(r);
    }
```

The response should only contain a JSON object. Postman confirms that that's actually what's returned by the web server.
What I get with cpp-netlib looks like this:

```
1cd0
{"http_code":200,"data":{"version":null,"requests":{"execution_time":""} /* .... */ } }
0
```

or like this:

```
f8f
{"http_code":200,"data":{"version":null,"requests":{ /* .... */ "2017-09-17 0
d41
9:45:34","updated_at":"2017-09-17 09:45:34"},{"id":"fcfe0d50-9b7b-11e7-ad59" /* ... */ ]}
0
```

The JSON objects are shortened. In the second example the JSON string is cut in the middle of a string by `d41`.

I have no idea where these artifacts are coming from. There are no other outputs and when I remove `std::cout << http::body(r)` the output is gone.

Boost version: 1.62.0
cpp-netlib version: 0.12.0
IDE: CLion 2017.2.2
OS: Kubuntu 17.04

**Comment by igorpeshansky** [comment]
I believe what you're seeing is [HTTP 1.1 chunked encoding](https://en.wikipedia.org/wiki/Chunked_transfer_encoding). This should be handled transparently by `http::body(r))`, so sounds like a bug. @deanberris, WDYT?

**Comment by deanberris** [comment]
That's correct -- this is fixed in 0.13.0-rc0. Can you try that version and see whether you still run into the problem?

**Comment by matthenning** [comment]
I was using this workaround until now: https://github.com/cpp-netlib/cpp-netlib/issues/110#issuecomment-5826644
I'll try 0.13.0-rc0 in the next few days and report back.
